src/cargo/util/toml.rs

```diff
@@ -551,26 +550,38 @@ impl TomlManifest {
             };
 
             // Collect the deps
-            try!(process_dependencies(&mut cx, self.dependencies.as_ref(),
-                                      None));
-            try!(process_dependencies(&mut cx, self.dev_dependencies.as_ref(),
-                                      Some(Kind::Development)));
-            try!(process_dependencies(&mut cx, self.build_dependencies.as_ref(),
-                                      Some(Kind::Build)));
-
-            if let Some(targets) = self.target.as_ref() {
-                for (name, platform) in targets.iter() {
-                    cx.platform = Some(try!(name.parse()));
-                    try!(process_dependencies(&mut cx,
-                                              platform.dependencies.as_ref(),
-                                              None));
-                    try!(process_dependencies(&mut cx,
-                                              platform.build_dependencies.as_ref(),
-                                              Some(Kind::Build)));
-                    try!(process_dependencies(&mut cx,
-                                              platform.dev_dependencies.as_ref(),
-                                              Some(Kind::Development)));
+
+            fn process_dependencies(
+                cx: &mut Context,
+                ds: [&Option<HashMap<String, TomlDependency>>; 3])
+                -> CargoResult<()>
+            {
+                let dep_iter = ds[0].iter().map(|d| (d, None))
+                    .chain(ds[1].iter().map(|d| (d, Some(Kind::Development))))
+                    .chain(ds[2].iter().map(|d| (d, Some(Kind::Build))))
```

Comment:
This seems a little magical to me, could the previous behavior of passing this in explicitly be kept instead?